### PR TITLE
Colin

### PR DESCRIPTION
**Pull request by ultrawide**
Added a general logging queue and a task that writes the logging queue items to file. 
https://github.com/SFUSatClub/obc-firmware/issues/9

One concern is the stack size when first creating logToFileTask in sfu_tasks_main.c.

### CODE REVIEW

File: platform-launchpad/source/sys_main.c [hunk 1]
```diff
@@ -142,8 +142,6 @@ int main(void)
 #pragma diag_suppress 112
 	/* USER CODE END */
 
-	/* USER CODE END */
-
```
Comment (liquiddandruff) [inline platform-launchpad/source/sys_main.c]: Revert these changes

File: platform-launchpad/halcogen.dil [hunk 2]
```diff
@@ -1,4 +1,4 @@
-# TMS570LS0432PZ 04/08/18 10:05:47
```
Comment (liquiddandruff) [inline platform-launchpad/halcogen.dil]: Revert these changes

File: SFUsat/sfu_rtc.h [hunk 3]
```diff
@@ -9,6 +9,7 @@
 #define SFUSAT_SFU_RTC_H_
 #include "FreeRTOS.h"
 #include "rtos_semphr.h"
+#include "spi.h"
```
Comment (liquiddandruff) [inline SFUsat/sfu_rtc.h]: Revert this (unless it's needed and wasn't added before for some reason)

File: platform-launchpad/source/sys_main.c (hunk 1 above)
Comment (ultrawide) [reply platform-launchpad/source/sys_main.c]: Done!

File: SFUsat/sfu_logging_task.c [hunk 4]
```diff
@@ -5,7 +5,8 @@
  *      Author: Colin
  */
 
-#include "sfu_logging_queue.h"
+#include <sfu_logging_task.h>
```
Comment (liquiddandruff) [inline SFUsat/sfu_logging_task.c]: Change to quotes

File: SFUsat/sfu_logging_task.c [hunk 5]
```diff
@@ -0,0 +1,47 @@
+/*
+ * sfu_logging_queue.c
```
Comment (liquiddandruff) [inline SFUsat/sfu_logging_task.c]: Again for consistency, the files should be renamed to `sfu_task_logging.c` etc, and comments such as these should be updated appropriately.

File: SFUsat/sfu_logging_task.h [hunk 6]
```diff
@@ -0,0 +1,47 @@
+/*
+ * sfu_logging_queue.h
```
Comment (liquiddandruff) [inline SFUsat/sfu_logging_task.h]: Fix name

File: SFUsat/sfu_logging_task.h [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+ *
+ *  Created on: Mar 27, 2018
+ *      Author: Colin Chan
+ *
+ *  See https://github.com/SFUSatClub/obc-firmware/issues/9
+ */
+
+#ifndef SFUSAT_ERROR_LOGGING_H_
```
Comment (liquiddandruff) [inline SFUsat/sfu_logging_task.h]: Should be `SFUSAT_TASK_LOGGING_H_`

File: SFUsat/sfu_task_main.c [hunk 8]
```diff
@@ -16,12 +16,10 @@
 #include "sfu_fs_structure.h"
 #include "flash_mibspi.h"
 #include "sfu_startup.h"
-
-//  ---------- SFUSat Tests (optional) ----------
 #include "sfu_triumf.h"
 #include "unit_tests/unit_tests.h"
 #include "examples/sfusat_examples.h"
-
+#include <sfu_logging_task.h>
```
Comment (liquiddandruff) [inline SFUsat/sfu_task_main.c]: Angle to quotes here too.
